core/rest-client/src/http_client.rs

```diff
@@ -432,4 +433,66 @@ mod tests {
 			Error::DeserializeParseError(err, String::from_utf8_lossy(encoded_body).to_string())
 		})
 	}
+
+	#[test]
+	fn get_from_site_with_self_signed_certificate_fails() {
+		let base_url = Url::parse("https://self-signed.badssl.com").unwrap();
+		let result = get_for_test_certificates(base_url);
+		assert_matches!(result, Err(Error::HttpReqError(_)));
+	}
+
+	#[test]
+	fn get_from_site_with_letsencrypt_isrgrootx1_valid_certificate_fails() {
+		let base_url = Url::parse("https://valid-isrgrootx1.letsencrypt.org").unwrap();
+		let result = get_for_test_certificates(base_url);
+		assert_matches!(result, Err(Error::HttpReqError(_)));
+	}
+
+	#[test]
+	fn get_from_site_with_letsencrypt_isrgrootx1_revoked_certificate_fails() {
+		let base_url = Url::parse("https://revoked-isrgrootx1.letsencrypt.org").unwrap();
+		let result = get_for_test_certificates(base_url);
+		assert_matches!(result, Err(Error::HttpReqError(_)));
+	}
+
+	#[test]
+	fn get_from_site_with_letsencrypt_isrgrootx1_expired_certificate_fails() {
+		let base_url = Url::parse("https://expired-isrgrootx1.letsencrypt.org").unwrap();
+		let result = get_for_test_certificates(base_url);
+		assert_matches!(result, Err(Error::HttpReqError(_)));
+	}
+
+	#[test]
+	fn get_from_site_with_letsencrypt_isrgrootx2_valid_certificate_fails() {
+		let base_url = Url::parse("https://valid-isrgrootx2.letsencrypt.org").unwrap();
+		let result = get_for_test_certificates(base_url);
+		assert_matches!(result, Err(Error::HttpReqError(_)));
+	}
+
+	#[test]
+	fn get_from_site_with_letsencrypt_isrgrootx2_revoked_certificate_fails() {
+		let base_url = Url::parse("https://revoked-isrgrootx2.letsencrypt.org").unwrap();
+		let result = get_for_test_certificates(base_url);
+		assert_matches!(result, Err(Error::HttpReqError(_)));
+	}
+
+	#[test]
+	fn get_from_site_with_letsencrypt_isrgrootx2_expired_certificate_fails() {
+		let base_url = Url::parse("https://expired-isrgrootx2.letsencrypt.org").unwrap();
+		let result = get_for_test_certificates(base_url);
+		assert_matches!(result, Err(Error::HttpReqError(_)));
+	}
+
+	fn get_for_test_certificates(base_url: Url) -> Result<(Response, EncodedBody), Error> {
```

Comment:
Rename suggestion:
`send_http_get_request`

Reply:
All other test are name "get" or "post", so I don't think I should change them. Or we change all test names

Reply:
I don't think the test function name and the name of a helper function have to match. Anyway, it was a suggestion only, not a change request, so the final decision is up to you :)